src/coreclr/inc/corhlprpriv.h

```diff
@@ -225,71 +225,33 @@ class CQuickMemoryBase
         iSize = cbTotal;
     }
 
-
-    // Convert UTF8 string to UNICODE string, optimized for speed
-    HRESULT ConvertUtf8_UnicodeNoThrow(const char * utf8str)
-    {
-        bool allAscii;
-        DWORD length;
-
-        HRESULT hr = FString::Utf8_Unicode_Length(utf8str, & allAscii, & length);
-
-        if (SUCCEEDED(hr))
-        {
-            LPWSTR buffer = (LPWSTR) AllocNoThrow((length + 1) * sizeof(WCHAR));
-
-            if (buffer == NULL)
-            {
-                hr = E_OUTOFMEMORY;
-            }
-            else
-            {
-                hr = FString::Utf8_Unicode(utf8str, allAscii, buffer, length);
-            }
-        }
-
-        return hr;
-    }
-
     // Convert UTF8 string to UNICODE string, optimized for speed
     void ConvertUtf8_Unicode(const char * utf8str)
     {
-        bool allAscii;
-        DWORD length;
+        size_t sourceLen = strlen(utf8str);
+        size_t destLen = minipal_get_length_utf8_to_utf16(utf8str, sourceLen, 0);
 
-        HRESULT hr = FString::Utf8_Unicode_Length(utf8str, & allAscii, & length);
+        CHAR16_T* buffer = (CHAR16_T*) AllocThrows((destLen + 1) * sizeof(CHAR16_T));
+        buffer[destLen] = W('\0');
 
-        if (SUCCEEDED(hr))
+        if (!minipal_convert_utf8_to_utf16(utf8str, sourceLen, buffer, destLen, 0))
         {
-            LPWSTR buffer = (LPWSTR) AllocThrows((length + 1) * sizeof(WCHAR));
-
-            hr = FString::Utf8_Unicode(utf8str, allAscii, buffer, length);
-        }
-
-        if (FAILED(hr))
-        {
-            ThrowHR(hr);
+            ThrowHR(EMAKEHR(ERROR_NO_UNICODE_TRANSLATION));
```

Comment:
```suggestion
            ThrowHR(EMAKEHR(errno));
```

it will map to errors 1:1.